sync/statesync/code_syncer.go

```diff
@@ -47,96 +49,67 @@ type CodeSyncerConfig struct {
 // codeSyncer syncs code bytes from the network in a seprate thread.
 // Tracks outstanding requests in the DB, so that it will still fulfill them if interrupted.
 type codeSyncer struct {
-	lock sync.Mutex
-
-	CodeSyncerConfig
+	db                       ethdb.Database
+	client                   statesyncclient.Client
+	maxOutstandingCodeHashes int
+	numCodeFetchingWorkers   int
 
+	lock                  sync.Mutex
 	outstandingCodeHashes set.Set[ids.ID]  // Set of code hashes that we need to fetch from the network.
```

Comment:
This isn't related to the PR, but this should really be
```suggestion
	outstandingCodeHashes set.Set[common.Hash]  // Set of code hashes that we need to fetch from the network.
```